src/test/fuzz/netmsg.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2020-2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <banman.h>
+#include <blockencodings.h>
+#include <consensus/consensus.h>
+#include <protocol.h>
+#include <streams.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/mining.h>
+#include <test/util/setup_common.h>
+#include <test/util/validation.h>
+#include <util/check.h>
+#include <util/time.h>
+#include <validationinterface.h>
+
+#include <chrono>
+#include <string>
+#include <thread>
+
+namespace {
+NetTestingSetup* g_setup;
+} // namespace
+
+static size_t& GetNumMsgTypes()
+{
+    static size_t g_num_msg_types{0};
+    return g_num_msg_types;
+}
+#define FUZZ_TARGET_MSG(msg_type)                          \
+    struct msg_type##CountBeforeMain {                     \
+        msg_type##CountBeforeMain()                        \
+        {                                                  \
+            ++GetNumMsgTypes();                            \
+        }                                                  \
+    } const static g_##msg_type##count_before_main;        \
+    FUZZ_TARGET_INIT(netmsg_##msg_type, initialize_netmsg) \
+    {                                                      \
+        fuzz_target(buffer, #msg_type);                    \
+    }
+
+void initialize_netmsg()
+{
+    Assert(GetNumMsgTypes() == getAllNetMessageTypes().size()); // If this fails, add or remove the message type below
+
+    static const auto testing_setup = MakeNoLogFileContext<NetTestingSetup>();
+    g_setup = testing_setup.get();
+    for (int i = 0; i < 2 * COINBASE_MATURITY; i++) {
+        MineBlock(g_setup->m_node, CScript() << OP_TRUE);
+    }
+    SyncWithValidationInterfaceQueue();
+}
+
+static void fuzz_target(FuzzBufferType buffer, const std::string& limit_to_message_type)
```

Comment:
Pushed an improvement that changes it to about 17 exec/s, it tends to increase if the test is left to run for some time.

IMO e2e fuzzing has some value.

Reply:
> IMO e2e fuzzing has some value.

Yes, but I think it should focus on what you are trying to test. If you keep the target to broad the fuzzer is gonna have a hard time finding interesting test cases (especially if the target is slow). One obvious way of focusing is to only fuzz smaller units of code and in the e2e case you probably want to narrow the way you are interpreting the fuzz input so that you are actually testing what you want to test.

The approach here currently seems to be very similar to the `process_message(s)` targets (which is why you suggested to replace them iiuc), in that you are interpreting most of the fuzz input as a sequence of arbitrary messages, so the questions that I have are:
* What do you want to test here? Message processing? connman internals? the interactions between connman/peerman?
* What bugs is your target capable of finding that `process_message(s)` (or other targets) are not? In other words which areas/interactions is this fuzzing that are previously not covered?
* Are the message contents (and their mutations) of all message types useful for what you are trying to test here? e.g. does it matter if a tx message contains a transaction with 1 input or 5? because the fuzzer will likely find that difference interesting but it doesn't seem like this would add value to what you are trying to test here, since the goal (correct me if I am wrong) isn't fuzzing validation.
* Could the input interpretation be narrowed to only include the messages that lead to interactions you are trying to test?

Reply:
After running it for some time, the maximum exec/s I got was 21 exec/s which is extremely low. It's not all the targets that reach more than 1000 exec/s here but 21 is too slow.

Reply:
"resolving" this thread because I removed the fuzz tests from this PR